crates/fluvio/src/producer/mod.rs

```diff
@@ -63,9 +103,19 @@ impl TopicProducer {
         K: Into<RecordKey>,
         V: Into<RecordData>,
     {
-        let record_key = key.into();
-        let record_value = value.into();
-        self.send_all(Some((record_key, record_value))).await?;
+        let key = key.into();
+        let value = value.into();
+        let record = Record::from((key, value));
+        let pending_record = PendingRecord {
+            partition: (),
+            record,
+            topic: self.topic.clone(),
+        };
+        let msg = DispatcherMessage::Record(pending_record);
+
+        self.inner.dispatcher.send(msg).await.map_err(|_| {
```

Comment:
So what happens to pending records?  Shouldn't this wait if producer msg is in flight?

Reply:
This will now wait whenever there are messages in flight. This channel is now a `bounded(1)`, which means that the sender must wait until the dispatcher is ready to receive and buffer a record. This only happens when the dispatcher is not busy waiting on producer responses.

Reply:
but channel is 1 sized that means it has 1 pending msg?

Reply:
It would be more clean if buffer management and communication to SPU are separated.  Buffer management can be done at time send to make it easier to manage state.

Reply:
Now using a rendesvous channel, so `dispatcher.send` will not complete until the dispatcher actually takes it from the receiver, ensuring there are no "pending" records in limbo.